File: tests/fixtures/go/sample.go
package sample

import (
	"fmt"
	"net/http"
)

type UserService struct {
	Name string
	Age  int
}

type Logger interface {
	Log(msg string)
	Error(msg string)
}

func NewUserService(name string) *UserService {
	return &UserService{Name: name}
}

func (s *UserService) GetName() string {
	fmt.Println(s.Name)
	return s.Name
}

func handleRequest(w http.ResponseWriter, r *http.Request) {
	svc := NewUserService("test")
	svc.GetName()
}

func classify(x int) string {
	if x > 0 {
		return "positive"
	}
	if x < 0 {
		return "negative"
	}
	return "zero"
}

File: tests/fixtures/ruby/sample.rb
class UserService
  def initialize(name)
    @name = name
  end

  def classify(x)
    if x > 0
      'positive'
    elsif x < 0
      'negative'
    else
      'zero'
    end
  end

  def process(items)
    results = []
    items.each do |item|
      case item
      when String
        results << item.upcase
      when Integer
        results << item.to_s
      else
        results << item.inspect
      end
    end
    results
  end
end

def helper(value)
  value ? value * 2 : 0
end
